### ISSUE TITLE
how effect on subset is calculated when clusters are specified?

### ISSUE THREAD

**Issue by ZhangMengxia** [grf-labs/grf#431]
When clusters are specified, if I want to know the ATE for X[,j]=a and X[,j]=b, how are they calculated?

Suppose within one cluster, some individuals have X[,j]=a while others have X[,j]=b.

Thanks for your time!

**Comment by swager** [comment]
The cluster-robust algorithm we use in GRF is described in Section 1.2 of https://arxiv.org/pdf/1902.07409.pdf. Let me know if this helps, or if you still have questions about it.

**Comment by ZhangMengxia** [comment]
@swager Thanks a lot for your time!

I think I got the idea of section 1.2 of this paper. Let me rephrase my question.

Using equation (8) of the same paper as an example, where i is student, j is school. We first compute cluster level mean treatment effect by ![CodeCogsEqn (6)](https://user-images.githubusercontent.com/17790770/59643496-c06da100-911d-11e9-8e38-2fbb02d870eb.gif)

Then the overall mean treatment effect is ![CodeCogsEqn (5)](https://user-images.githubusercontent.com/17790770/59643201-92d42800-911c-11e9-83d7-a3dd2a3f4cfb.gif)


Now I want the mean treatment effect for a subsample. For example, by setting subset = X[,j]==1 in the average_partial_effect() function. Using the example of schools, 

1. Suppose some schools are public schools and others are private schools. If we want to calculate the mean treatment effect of the intervention for the subsample of public schools, shall we use the following equation? 
![CodeCogsEqn (3)](https://user-images.githubusercontent.com/17790770/59643127-64564d00-911c-11e9-89c8-5de2e2963092.gif) , where j belongs to public schools.

2. Another situation is, for example, I want the mean treatment effect for female students. In this case, a cluster has both female students and male students, how to calculate the mean treatment effect on this subsample?

**Comment by swager** [comment]
Ok thanks for the clarification; now I understand. For case (1), if we want an average over a subset of clusters, then yes -- the estimand is clear, and it's what you describe.

For (2), what we do algorithmically is that we compute sample weights as though we wanted to do an ATE, and literally just subset the weights, see:

https://github.com/grf-labs/grf/blob/15ee993c8ab63da8d6c3c61e4f309d8ae02ca222/r-package/grf/R/average_treatment_effect.R#L102

It seems like there may be room to do something more principled here -- let me know if you have any thoughts about it.

**Comment by ZhangMengxia** [comment]
@swager Thanks for the explanation!

Do you mean in the second case, each student get the same weight? 

And the mean treatment effect in this subsample is the simple average of the individual-specific effects across all female students?

**Comment by swager** [comment]
In this case, if we write Nj for the number of students in each cluster, Njf for the number of female students in the cluster, and Tjf for the estimated average treatment effect for female students in cluster j, then we end up returning the estimate

T = sum_j Tjf Njf  / Nj / (sum_j Njf / Nj).

Granted, it's a somewhat odd estimand -- on the other hand, it's not obviously clear to me that there's an obvious answer as to how one should want to do the weighting here. Is there some weighting you'd want in your application?

**Comment by ZhangMengxia** [comment]
@swager  thanks a lot for your answers! 

No, I do not have a specific weighting in my mind. I just want to know the estimand in this case. Your answers help a lot!

**Comment by swager** [comment]
In the latest (1.0) release, we have simplified the default approach to weighting by clusters (see #488 and #545). The default behavior is to give each sample equal weight (regardless of cluster size). If a different weighting scheme is required, it can be specified via the `sample.weights` argument.
